### ISSUE TITLE
Custom <modules> section

### ISSUE THREAD

**Issue by m1ga** [prop/ti.multiconfiguration#3]
Would it be possible to change the modules section too?

```js
modules: [{name: "name", "version": "1.0.0", "platform": "android"}]
```
and output
```xml
<modules>
  <module version="1.0.0" platform="android">name</module>
<modules>
```

**Comment by drauggres** [comment]
Hi there.
IIRC there was a problem with specifying modules in the config this way. Or maybe I was too lazy to implement this and used `"raw-config"`, it should be possible to add modules inside it. `"raw-config"` should have the same stucture as the internal `build` configuration object (upd: you can inspect it in debugger, see below), e.g.:
```json
      "raw-config": {
        "tiapp": {
          "ios":{
            "use-app-thinning": true,
            "min-ios-ver": 8,
            "enable-launch-screen-storyboard": true,
            "default-background-color": "#ffffff"
          }
        }
      },
```

If you want to debug this (or any other ti cli plugin):
1. add `debugger` inside the hook, something like: 
```javascript
  // first argument is the build configuration object mentioned above 
  cli.addHook('build.pre.construct', function(build, finished) {
    debugger;
```
2. run build:
```shell
node --inspect-brk /usr/local/lib/node_modules/titanium/bin/titanium build -p android --build-profile customer1 --build-only
```

Basically it should be same old `ti build ...`, but with full path to the `titanium` script and prepended  with `node --inspect-brk`

3. Open dev tools in a chromium based browser, click "node.js" icon
4. Enjoy debugging :-)
